### PR TITLE
DOC: Add tags infrastructure for gallery examples

### PR DESCRIPTION
**Pull request by melissawm**
## PR summary

Adds [sphinx-tags](https://sphinx-tags.readthedocs.io/en/latest/index.html) as a dependency and some basic configuration to display tags with color, using [sphinx-design](https://sphinx-design.readthedocs.io/en/latest/).

This is a proof-of-concept implementation of tags for the gallery of examples. The actual tags will be the result of the Google Season of Docs project by @esibinga.

This PR is mostly a demo of how this would work in practice, and feedback is appreciated.

Related: #26851 

There is no release note but this may need one when it is ready to merge.

cc @story645 

Previews:

- [Example gallery entry page, with the link to all tags](https://output.circle-artifacts.com/output/job/63217eb6-7b19-4d1f-b35a-9aa17e8da975/artifacts/0/doc/build/html/gallery/index.html)
- [Tagged animation example](https://output.circle-artifacts.com/output/job/63217eb6-7b19-4d1f-b35a-9aa17e8da975/artifacts/0/doc/build/html/gallery/animation/animated_histogram.html) (**scroll to the bottom to see the tags**)
- [Tagged example](https://output.circle-artifacts.com/output/job/63217eb6-7b19-4d1f-b35a-9aa17e8da975/artifacts/0/doc/build/html/gallery/event_handling/resample.html) (**scroll to the bottom to see the tags**)

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by jklymak** [comment]
This seems good.  Thanks for mocking it up.  There are a numnber of suggestions for the layout etc that are probably sphinx-tags things?  

1. I wonder if the colors should be defined in the tags file somehow?

2. It's a shame that the tag page isn't visual.  

3. The breadcrumbs don't seem to be working?  

Does this work by making a bunch of static index pages?  

Specific to the PR I wonder if a better way than Note can be used to indicate the tags index?

**Comment by QuLogic** [comment]
Are we able to use spaces in the tags? At least after the colon?

**Comment by melissawm** [comment]
Yes @QuLogic - thanks to @JWCook we can now do that 😄 

I will cut a release of sphinx-tags in the next day or two and will update this PR accordingly.

**Comment by melissawm** [comment]
Alright folks - just released sphinx-tags v0.3 and will mark this as ready for review. To answer a few of the questions above:

> Does this work by making a bunch of static index pages?
Yes, it does. There may be better solutions but it generates pages on build-time, so no need to store them in the repo.

> Specific to the PR I wonder if a better way than Note can be used to indicate the tags index?
I am happy to move this to somewhere else - my thinking was because this is still a prototype we may not want to make this a huge feature while the tags are not at least 50-70% done? 

> I wonder if the colors should be defined in the tags file somehow?

Because these files are generated at build time, I'm not sure we could do that without significant rework.

> It's a shame that the tag page isn't visual.

Agreed! Definitely something to consider for a future release of sphinx-tags.

> wondering if we can do a [animation] tag that pulls every [animation:] sub example?

Also something we're tracking on sphinx-tags. Would not be able to pull this off for this release though.

---

@story645 since it would be useful to have this for the sprint at PyData NYC tomorrow, I'll be prioritizing any further comments here. For now, I'm marking as ready for review.

File: doc/_static/mpl.css [hunk 1]
```diff
@@ -3,6 +3,11 @@
     --pst-color-link-hover: var(--pst-color-secondary);
     --sd-color-primary: var(--pst-color-primary);
     --sd-color-primary-text: var(--pst-color-text-base);
+    --sd-color-secondary: #ee9040;
+    --sd-color-success: #28a745;
+    --sd-color-dark: #323232;
+    --sd-color-danger: #dc3545;
+    --sd-color-light: #c9c9c9;
```
Comment (QuLogic) [inline doc/_static/mpl.css:10]: Where do these colours come from? Are there not any variable in pydata-sphinx-theme that correspond to them?

File: galleries/examples/animation/animated_histogram.py [hunk 2]
```diff
@@ -55,3 +55,6 @@ def animate(frame_number):
 ani = animation.FuncAnimation(fig, prepare_animation(bar_container), 50,
                               repeat=False, blit=True)
 plt.show()
+
+# %%
+# .. tags:: plot-type:histogram, animation
```
Comment (QuLogic) [inline galleries/examples/animation/animated_histogram.py:60]: Assuming the spaces now work:
```suggestion
# .. tags:: plot-type: histogram, animation
```

File: galleries/examples/animation/multiple_axes.py [hunk 3]
```diff
@@ -80,3 +80,5 @@ def animate(i):
 #
 #    - `matplotlib.patches.ConnectionPatch`
 #    - `matplotlib.animation.FuncAnimation`
+#
+# .. tags:: component:axes, animation
```
Comment (QuLogic) [inline galleries/examples/animation/multiple_axes.py:84]:
```suggestion
# .. tags:: component: axes, animation
```

File: galleries/examples/event_handling/resample.py [hunk 4]
```diff
@@ -75,3 +75,6 @@ def update(self, ax):
 ax.callbacks.connect('xlim_changed', d.update)
 ax.set_xlim(16, 365)
 plt.show()
+
+# %%
+# .. tags:: interactivity:zoom, event-handling
```
Comment (QuLogic) [inline galleries/examples/event_handling/resample.py:80]:
```suggestion
# .. tags:: interactivity: zoom, event-handling
```

File: requirements/doc/doc-requirements.txt [hunk 5]
```diff
@@ -21,3 +21,4 @@ sphinxcontrib-svg2pdfconverter>=1.1.0
 sphinx-gallery>=0.12.0
 sphinx-copybutton
 sphinx-design
+sphinx-tags
```
Comment (QuLogic) [inline requirements/doc/doc-requirements.txt:24]: This should get a minimum for the release with spaces in it.

**Comment by melissawm** [comment]
Good point on the normalizing.

And yes, colors can be pulled from variables, I set them manually as a proof of concept. I can try and make them more general tomorrow, although I wasn't even suggesting these should be the final colors, these were more to give the idea of how they could look. We can think more deeply about these colors and accessibility, for example.

I'm on mobile right now but can take a look at these things first in the morning.

**Comment by melissawm** [comment]
> Guess CI isn't complaining about the tag so

I'm not sure what you mean, are you having issues building?

**Comment by melissawm** [comment]
I am not sure what's the deal with appveyor, but the docs are building ok: 

- Admonition about tagging: https://output.circle-artifacts.com/output/job/5c749464-1398-4efa-b058-4b651d159ac9/artifacts/0/doc/build/html/gallery/index.html

- Tagged example: https://output.circle-artifacts.com/output/job/5c749464-1398-4efa-b058-4b651d159ac9/artifacts/0/doc/build/html/gallery/animation/multiple_axes.html

**Comment by tacaswell** [comment]
Lets kick the tires on this at the pydataNYC sprint today!

**Pull request merged**